File: packages/code-highlighting/android/src/main/java/com/apollohg/editor/highlighting/NativeCodeHighlightingModule.kt
package com.apollohg.editor.highlighting

import com.apollohg.editor.CodeHighlightingRegistry
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class NativeCodeHighlightingModule : Module() {
    private val provider = SyntectHighlightingProvider()

    override fun definition() = ModuleDefinition {
        Name("NativeCodeHighlighting")
        Function("initialize") {
            CodeHighlightingRegistry.register(provider)
            provider.version
        }
    }
}

File: packages/code-highlighting/ios/NativeCodeHighlightingModule.swift
@preconcurrency public import ExpoModulesCore
import ReactNativeProseEditor

public final class NativeCodeHighlightingModule: Module {
    private let provider = SyntectHighlightingProvider()

    public func definition() -> ModuleDefinition {
        Name("NativeCodeHighlighting")
        Function("initialize") { () throws -> Int in
            try NativeCodeHighlightingRegistry.register(provider: self.provider)
            return self.provider.version
        }
    }
}
